### PR TITLE
Add Heavy-ball momentum, Nesterov momentum, AdamW, and NAdamW baselines

### PR DESCRIPTION
**Pull request by runame**
Addresses #287, #288, #289, and #290. I chose the directory names based on the target-setting runs naming scheme.

**TODO**

- [ ] Make adjustments based on the answer to [this question](https://github.com/mlcommons/algorithmic-efficiency/issues/287#issuecomment-1407416305).
- [ ] Add hyperparameter tuning search spaces. I don't think I have access to them, maybe someone can just replace/fill in the placeholder files I have added.
- Add instructions on how to run each submission in a README, but this can only be done when the API allows for valid submissions.

### CODE REVIEW

File: baselines/adamw/pytorch/submission.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      workload.metrics_logger.append_scalar_metrics(
+          {
+              'loss': loss.item(),
+              'grad_norm': grad_norm.item(),
+          }, global_step)
+    logging.info('%d) loss = %0.3f, grad_norm = %0.3f',
+                 global_step,
+                 loss.item(),
+                 grad_norm.item())
+
```
Comment (znado) [inline baselines/adamw/pytorch/submission.py:118]: Do you think it makes sense to add `optimizer_state[‘optimizer’].zero_grad(set_to_none=True)` to the line before the `return` in all the pytorch baselines? That should more aggressively free the memory being used by grads. I'm no pytorch pro though so I'll leave it up to you

**Review by znado — APPROVED** [review]

File: baselines/adamw/pytorch/submission.py (hunk 1 above)
Comment (runame) [reply baselines/adamw/pytorch/submission.py:118]: Cool idea, I haven't really seen this anywhere together with setting the gradients to 0 before the backward pass. This could in some cases maybe lead to a small slow-down because memory has to be allocated again before each update, but apparently [not always](https://discuss.pytorch.org/t/freeing-gradients-memory-after-optimizer-step/116346/4). Do we actually care about saving memory in the case of the baselines? They run with the fixed batch sizes anyway.

Comment (znado) [reply baselines/adamw/pytorch/submission.py:118]: yeah that makes sense, I was playing around with it for criteo but that was it. fine with leaving it out for now.
